doc/src/modules/solvers/solveset.rst

```diff
@@ -1,12 +1,463 @@
+   iteratively.
+
+ * If none of the above method is successful, then method of polynomial is
+   used as follows:
+
+   - ``\_solve\_as\_rational`` is called, it's third argument is the
+     ``solveset\_solver`` which can either be ``solveset_real`` or
+     ``solveset\_complex`` based on these respective poly solvers
+     ``\_solve\_as\_poly\_real`` and ``\_solve\_as\_poly\_complex`` is called.
+
+   - The underlying method ``\_solve\_as\_poly`` solves the equation using
+     polynomial techniques if it already is a polynomial equation or, with
+     a change of variables, can be made so.
+
+ * The Final solution set obtained is taken intersection with the input
+   domain, and the resultant solution is returned.
+
+.. Remember to change the above part when the new solver is implemented.
+
+
+How do we manipulate and return an infinite solution?
+-----------------------------------------------------
+
+ * In Real Domain, we use our ``ImageSet`` class in the sets module to
+   return infinite solutions. ``ImageSet`` is an Image of a set under
+   a mathematical function. For example, to represent the solution
+   of the equation `sin(x) == 0`, we can use the ImageSet as:
+
+   ``ImageSet(Lambda(n, 2*n*pi), Integers())``
+
+   Where n is a dummy variable. It is basically the image of the
+   Integers set under the function `2*n*pi`.
+
+ * In Complex Domain, we use Complex Sets, which is implemented as
+   ``ComplexPlane`` class in the sets module, to represent infinite
+   solution in the argand plane. For example to represent the solution
+   of the equation `|z| == 1`, which is a unit circle, we can use
+   the ComplexPlane as:
+
+   ``ComplexPlane(FiniteSet(1)*Interval(0, 2*pi), polar=True)``
+
+   Where the ``FiniteSet`` in the ``ProductSet`` is the range of the value of `r`,
+   which is the radius of the circle and the ``Interval`` is the range of theta,
+   representing a unit circle in the argand plane.
+
+   Note: We also have non-polar form notation for representing solution
+   in rectangular form. For example, to represent first quadrant in argand
+   plane, we can write the ComplexPlane as:
+
+   ``ComplexPlane(Interval(0, oo)*Interval(0, oo))``
+
+   where the Intervals are the range of `x` and `y` for the set of complex
+   number `(x + I*y)`.
+
+
+How does solveset ensures that it is not returning any wrong solution?
+----------------------------------------------------------------------
+
+ Solvers in a Computer Algebra System are based on heuristics algorithms,
+ so it's usually very hard to ensure cent percent correctness, in every
+ possible case. However there are still a lot of cases where we can ensure
+ correctness. Solveset tries to verify correctness wherever it can for
+ example:
+
+ Consider the equation `|x| = n`, a naive method to solve this equation
+ would return {-n, n} as it's solution, which is not correct since {-n, n}
+ can only be it's solution if and only if n is positive. Solveset returns
```

Comment:
"can be its solution if and only if \`\`n\`\` is positive" (remove redundant "only", backquote "n")